File: kyw0716/level0/특이한 정렬.js
function solution(numlist, n) {
  numlist.sort((a, b) => {
    const A = Math.abs(n - a);
    const B = Math.abs(n - b);
    return A > B ? 1 : A === B ? b - a : -1;
  });
  return numlist;
}
// https://school.programmers.co.kr/learn/courses/30/lessons/120880

File: sk000801/입문/특이한 정렬.java
import java.util.*;
class Solution {
    public int[] solution(int[] numlist, int n) {
        int[] answer = new int[numlist.length];
        List<Integer> special = new ArrayList<>();
        for(int i = 0; i< numlist.length; i++) {
            special.add(numlist[i]);
        }
        special.sort(new Comparator<Integer>() {
            public int compare(Integer a, Integer b) {
                int aa = Math.abs(a-n);
                int bb = Math.abs(b-n);
                
                if(aa == bb) return b-a;
                else return aa-bb;
            }
        });
        for(int i = 0; i< numlist.length; i++) {
            answer[i] = special.get(i);
        }
        return answer;
    }
}

